include/wil/networking.h

```diff
@@ -0,0 +1,1487 @@
+                extensionGuid = getacceptexsockaddrs_guid;
+                functionPtr = &table.f.GetAcceptExSockaddrs;
+                break;
+            }
+            case 4:
+            {
+                constexpr GUID transmitfile_guid = WSAID_TRANSMITFILE;
+                extensionGuid = transmitfile_guid;
+                functionPtr = &table.f.TransmitFile;
+                break;
+            }
+            case 5:
+            {
+                constexpr GUID transmitpackets_guid = WSAID_TRANSMITPACKETS;
+                extensionGuid = transmitpackets_guid;
+                functionPtr = &table.f.TransmitPackets;
+                break;
+            }
+            case 6:
+            {
+                constexpr GUID wsarecvmsg_guid = WSAID_WSARECVMSG;
+                extensionGuid = wsarecvmsg_guid;
+                functionPtr = &table.f.WSARecvMsg;
+                break;
+            }
+            case 7:
+            {
+                constexpr GUID wsasendmsg_guid = WSAID_WSASENDMSG;
+                extensionGuid = wsasendmsg_guid;
+                functionPtr = &table.f.WSASendMsg;
+                break;
+            }
+
+            default:
+                FAIL_FAST();
+            }
+
+            constexpr DWORD controlCode{SIO_GET_EXTENSION_FUNCTION_POINTER};
+            constexpr DWORD bytes{sizeof(void*)};
+            DWORD unused_bytes;
+            if (::WSAIoctl(
+                    localSocket.get(),
+                    controlCode,
+                    &extensionGuid,
+                    DWORD{sizeof(extensionGuid)},
+                    functionPtr,
+                    bytes,
+                    &unused_bytes,
+                    nullptr,
+                    nullptr) != 0)
+            {
+                const auto gle = ::WSAGetLastError();
+                LOG_IF_WIN32_ERROR(gle);
+
+                // if any failed to be found, something is very broken
+                // all should load, or all should fail
+                ::memset(&table.f, 0, sizeof(table.f));
+                successfully_loaded = false;
+            }
+        }
+
+        return table;
+    }
+
+    //
+    // definitions for rio_extension_function_table
+    //
+    inline rio_extension_function_table rio_extension_function_table::load() WI_NOEXCEPT
+    {
+        rio_extension_function_table table{};
+        // if WSAStartup failed, immediately exit
+        if (!table.wsa_reference_count)
+        {
+            return table;
+        }
+
+        // we need a temporary socket for the IOCTL to load the functions
+        const ::wil::unique_socket localSocket{::socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP)};
+        if (INVALID_SOCKET == localSocket.get())
+        {
+            return table;
+        }
+
+        constexpr DWORD controlCode{SIO_GET_MULTIPLE_EXTENSION_FUNCTION_POINTER};
+        constexpr DWORD bytes{sizeof(table.f)};
+        GUID rioGuid = WSAID_MULTIPLE_RIO;

```

Comment:
NIT: `GUID rioGuid{ WSAID_MULTIPLE_RIO };`

[ES.23: Prefer the {}-initializer syntax](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#es23-prefer-the--initializer-syntax)

Same comment applies throughout. Usually done, this leaped at given the 2 immediately previous lines do it but this line didn't. Any other like cases should be equally updated